sandbox/host/src/sandbox/wasmer_backend/cache.rs

```diff
@@ -0,0 +1,211 @@
+
+    let res = fs_cache.store(code_hash, &serialized_module);
+    log::trace!("Store module in FS cache with result: {:?}", res);
+}
+
+/// Altered copy of the `FileSystemCache` struct from `wasmer_cache` crate.
+#[derive(Debug, Clone)]
+pub struct FileSystemCache {
+    path: PathBuf,
+    ext: Option<String>,
+}
+
+impl FileSystemCache {
+    /// Construct a new `FileSystemCache` around the specified directory.
+    fn new<P: Into<PathBuf>>(path: P) -> io::Result<Self> {
+        let path: PathBuf = path.into();
+        if path.exists() {
+            let metadata = path.metadata()?;
+            if metadata.is_dir() {
+                if !metadata.permissions().readonly() {
+                    Ok(Self { path, ext: None })
+                } else {
+                    // This directory is readonly.
+                    Err(io::Error::new(
+                        io::ErrorKind::PermissionDenied,
+                        format!("the supplied path is readonly: {}", path.display()),
+                    ))
+                }
+            } else {
+                // This path points to a file.
+                Err(io::Error::new(
+                    io::ErrorKind::PermissionDenied,
+                    format!(
+                        "the supplied path already points to a file: {}",
+                        path.display()
+                    ),
+                ))
+            }
+        } else {
+            // Create the directory and any parent directories if they don't yet exist.
+            let res = fs::create_dir_all(&path);
+            if res.is_err() {
+                Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    format!("failed to create cache directory: {}", path.display()),
+                ))
+            } else {
+                Ok(Self { path, ext: None })
+            }
+        }
+    }
+
+    /// Load the serialized module from the cache.
+    fn load(&self, key: Hash) -> Result<Vec<u8>, io::Error> {
+        let filename = if let Some(ref ext) = self.ext {
+            format!("{}.{}", key, ext)
+        } else {
+            key.to_string()
+        };
+        let path = self.path.join(filename);
+
+        fs::read(path)
+    }
+
+    /// If an error occurs while deserializing then we can not trust it anymore
+    /// so delete the cache file
+    fn remove_key(&self, key: Hash) {
+        let filename = if let Some(ref ext) = self.ext {
+            format!("{}.{}", key, ext)
+        } else {
+            key.to_string()
+        };
+        let path = self.path.join(filename);
+
+        let _ = fs::remove_file(path);
+    }
+
+    /// Store the serialized module in the cache.
+    fn store(&mut self, key: Hash, serialized_module: &[u8]) -> Result<(), io::Error> {
+        let filename = if let Some(ref ext) = self.ext {
+            format!("{}.{}", key, ext)
+        } else {
+            key.to_string()
+        };
```

Comment:
extract standalone utility function

```suggestion
fn compose_filename(extension: Option<&String>, key: String) -> String {
     match extension {
         Some(extension) => format!("{key}.{extension}"),
         None => key,
      }
}

        let filename = compose_filename(self.ext.as_ref(), key.to_string());
```

Reply:
Extracted